### PR TITLE
Fix publishToSonatype Task in Release Action

### PR DESCRIPTION
**Pull request by adellinekania**
- Update `build.gradle` to ensure successful execution of `publishToSonatype` and `closeAndReleaseSonatypeStagingRepository` tasks. This change addresses previous validation rule violations.
- Adjust the artifact pattern.

### CODE REVIEW

**Review by pgorbachev — DISMISSED** [review]

File: build.gradle [hunk 1]
```diff
@@ -17,35 +19,39 @@ repositories {
 dependencies {
     implementation "com.github.node-gradle:gradle-node-plugin:${project.'gradle-node-plugin.version'}"
     api "org.octopusden.octopus.infrastructure:components-registry-service-client:${project['octopus-components-registry-service.version']}"
+    api "com.platformlib:platformlib-process-local:${project['platformlib-process.version']}"
```
Comment (CamaradeRoman) [inline build.gradle:22]: @adellinekania why this dependency has been added?

File: build.gradle [hunk 2]
```diff
@@ -57,50 +63,45 @@ nexusPublishing {
...
-    def signingKey = project.findProperty('signingKey')
-    def signingPassword = project.findProperty('signingPassword')
-
-    useInMemoryPgpKeys(signingKey, signingPassword)
-
-    gradle.taskGraph.whenReady { taskGraph ->
-        if (taskGraph.hasTask(':publishToSonatype')) {
-            sign publishing.publications.mavenJava
-        }
+if (project.hasProperty("nexus") && !project.version.toString().toUpperCase().endsWith("SNAPSHOT")) {
```
Comment (CamaradeRoman) [inline build.gradle:100]: @adellinekania I consider that as a bad solutions.

Why not https://github.com/octopusden/octopus-teamcity-automation/blob/08571ae1282e2605315eb580413b1afc2718d486/build.gradle.kts#L156 ?

**Review by CamaradeRoman — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (adellinekania) [reply build.gradle:22]: This is not a new dependency; it was already in use but listed under a different section. I have now merged it into only one dependency section.

File: build.gradle (hunk 2 above)
Comment (adellinekania) [reply build.gradle:100]: Okay, will do

File: build.gradle [hunk 3]
```diff
@@ -1,13 +1,15 @@
-//
 plugins {
     id 'groovy'
     id 'java-gradle-plugin'
+    id 'maven-publish'
     id 'io.github.gradle-nexus.publish-plugin'
     id 'signing'
 }
 
+version = project.findProperty('buildVersion') ?: System.getenv("BUILD_VERSION") ?: '1.0-SNAPSHOT'
```
[the remark this answers is not in the document]
Comment (adellinekania) [reply build.gradle]: It turns out the change isn't necessary. I've already updated it accordingly. Thank you for checking!

**Review by pgorbachev — APPROVED** [review]

**Review by CamaradeRoman — APPROVED** [review]

**Pull request merged**